tests/integration/istio_test.go

```diff
@@ -0,0 +1,496 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1alpha1 "maistra.io/istio-operator/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"istio.io/istio/pkg/ptr"
+)
+
+var _ = Describe("Istio resource", Ordered, func() {
+	const (
+		istioName         = "test-istio"
+		istioNamespace    = "istio-test"
+		workloadNamespace = "istio-test-workloads"
+
+		istioVersion = "latest"
+		oldVersion   = "v1.20.0"
+		newVersion   = "v1.20.1" // TODO: get these versions from versions.yaml
+
+		gracePeriod = 30 * time.Second
+		pilotImage  = "maistra.io/test:latest"
+	)
+	istioKey := client.ObjectKey{Name: istioName}
+	istio := &v1alpha1.Istio{}
+
+	SetDefaultEventuallyTimeout(30 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+
+	SetDefaultConsistentlyDuration(30 * time.Second)
```

Comment:
This long duration and the usage of `Consistently()` in 6 places really prolongs the tests... Currently this PR adds 5m(!) of execution time to `make test`. If we strip this duration down to `3s`, it is "only" 3m of additional execution time- which IMO is still too much and we should look into shortening it further.

Do we even really need to use `Consistently`? I only noticed this problem because during test execution the operator would be idle for 30s exactly 6 times (the amount of times we call `Consistently()`).

Reply:
Unfortunately, we do, because we're checking that the reconciliation does not happen. There's no other way to confirm that the operator won't do something. The duration needs to be long enough for us to be reasonably sure that the fact that nothing has happened is a result of the operator actually not doing it and not because something caused the operator to stall. 

However, given that a reconciliation typically takes around 2 to 3 seconds, we can probably shorten the duration from 30s to maybe 10s? I don't think we can shorten it to anything less without affecting our confidence in the test.

Reply:
The change to 10s sounds like a good compromise. We can then see how we can improve testing time in the future. Thanks Marko